paasta_tools/tron_tools.py

```diff
@@ -291,6 +304,10 @@ def _build_spark_config(self) -> Dict[str, str]:
             "spark.kubernetes.executor.label.paasta.yelp.com/cluster": self.get_cluster(),
             "spark.kubernetes.node.selector.yelp.com/pool": self.get_pool(),
             "spark.kubernetes.executor.label.yelp.com/pool": self.get_pool(),
+            # this relies on the PaaSTA workload contract being followed - $PAASTA_POD_IP
+            # will be, drumroll please, the routable IP for the Pod - this allows us to
+            # not need to worry about DNS
+            "spark.driver.host": "$PAASTA_POD_IP",
```

Comment:
Cool, I was a little confused here since I didn't know _build_spark_config would just be used to generate the actual command being run in the container, and therefore envvar expansion would work.

Is that always going to be the case?

Reply:
it should always be the case for now - so far, i think the only thing in spark that can be configured with yaml is some 'advanced' k8s knobs like pod affinity (but if the driver is ever configured like that, then that'll be even simpler than this somewhat magic approach)